### PR TITLE
Support to configure expose strategy and allow IP list for API server

### LINKED ISSUE #5141: Support for expose strategy and Allowed IP source ranges
**What would you like to be added? / User Story**:

For feature [#10407](https://github.com/kubermatic/kubermatic/issues/10407) source IP range setup for user clusters, below changes are needed:

1. Under **Cluster - Advanced Network Settings**, add a drop down box to select "Expose Strategy" . The values are: `NodePort, LoadBalancer and Tunneling`.
2. Field must be editable in run-time too (i.e., even after cluster creation)
3. As of now this feature works only for `LoadBalancer` strategy
4. When expose strategy is set as `LoadBalancer`, user should be able to configure source IP ranges like pod cidrBlocks
5. `Spec.APIServerAllowedIPRanges` and `spec.ExposeStrategy` are the respective fields in the cluster api.

NOTE: API endpoint NetworkDefaults is updated as part of [#5198](https://github.com/kubermatic/dashboard/pull/5198/files)

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
Signed-off-by: Waleed Malik <ahmedwaleedmalik@gmail.com>

**What this PR does / why we need it**:
Support for configuring expose strategy and API server allow list for clusters.

**Which issue(s) this PR fixes**:

Fixes #5141

**What type of PR is this?**

/kind feature

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Support for configuring expose strategy and API server allow list for clusters.
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: modules/web/src/app/wizard/step/cluster/component.ts [hunk 1]
```diff
@@ -211,6 +218,13 @@ export class ClusterStepComponent extends StepBase implements OnInit, ControlVal
       [Controls.IPv4CIDRMaskSize]: this._builder.control(null),
       [Controls.IPv6CIDRMaskSize]: this._builder.control(null),
       [Controls.NodeLocalDNSCache]: this._builder.control(false),
+      [Controls.ExposeStrategy]: this._builder.control(null),
+      [Controls.IPv4APIServerAllowedIPRanges]: this._builder.control(this._defaultAllowedIPRange, [
+        IPV4_CIDR_PATTERN_VALIDATOR,
+      ]),
+      [Controls.IPv6APIServerAllowedIPRanges]: this._builder.control(this._defaultAllowedIPv6Range, [
+        IPV6_CIDR_PATTERN_VALIDATOR,
+      ]),
```
Comment (Waseem826) [inline modules/web/src/app/wizard/step/cluster/component.ts:227]: Do we need `_dualStackRequiredIfValidator` here as well similar to Services CIDR and Allowed IP Range controls? Or is one empty value allowed in `apiServerAllowedIPRanges.cidrBlocks` array?

File: modules/web/src/app/wizard/step/cluster/template.html [hunk 2]
```diff
@@ -113,8 +113,23 @@
...
+              </mat-option>
+            </mat-select>
+            <mat-hint>Leave empty to use the default value. Check the documentation for more information&nbsp;<a target="_blank"
+                 rel="noopener"
+                 href="https://docs.kubermatic.com/kubermatic/main/tutorials-howtos/networking/expose-strategies/#configure-the-expose-strategy">here</a>.
+            </mat-hint>
+          </mat-form-field>
           <div fxLayout="row"
-               fxLayoutGap="50px"
+               fxLayoutGap="100px"
```
Comment (Waseem826) [inline modules/web/src/app/wizard/step/cluster/template.html:132]: I think `100px` is too much space or is there a specific reason to increase this?

**Review by Waseem826 — COMMENTED** [review]
Couple of comments. PTAL

File: modules/web/src/app/wizard/step/cluster/component.ts (hunk 1 above)
Comment (ahmedwaleedmalik) [reply modules/web/src/app/wizard/step/cluster/component.ts:227]: One empty value should be allowed in this case.

File: modules/web/src/app/wizard/step/cluster/template.html (hunk 2 above)
Comment (ahmedwaleedmalik) [reply modules/web/src/app/wizard/step/cluster/template.html:132]: It's to facilitate the `mat-hint` 50px gap was too low for it. Although we can use padding as an alternate as well.

**Review by Waseem826 — APPROVED** [review]

**Review by Waseem826 — APPROVED** [review]

**Pull request merged**